src/crawlers/cheerio_crawler.js

```diff
@@ -354,14 +384,22 @@ class CheerioCrawler {
      *
      * @ignore
      */
-    async _handleRequestFunction({ request, autoscaledPool }) {
+    async _handleRequestFunction({ request, autoscaledPool, session }) {
         if (this.prepareRequestFunction) await this.prepareRequestFunction({ request });
```

Comment:
Maybe `prepareRequestFunction` should also get access to session? Might wanna do something with cookies or stuff.

Reply:
That is true for example pass the auth header from the session.